test/tools/runner/hooks/configuration.js

```diff
@@ -95,15 +102,42 @@ before(async function () {
 
   this.configuration = new TestConfiguration(MONGODB_URI, context);
   await client.close();
-});
-
-// ensure all mock connections are closed after the suite is run
-after(() => mock.cleanup());
 
-// optionally enable test runner-wide plugins
-require('./plugins/deferred');
-require('./plugins/session_leak_checker');
-require('./plugins/client_leak_checker');
+  const currentEnv = {
+    topology: this.configuration.topologyType,
+    version: this.configuration.buildInfo.version,
+    node: process.version,
+    os: process.platform,
+    serverless: process.env.SERVERLESS === '1',
+    auth: process.env.AUTH === 'auth',
+    tls: process.env.SSL === 'ssl',
+    csfle: this.configuration.clientSideEncryption.enabled,
+    serverApi: MONGODB_API_VERSION,
+    atlas: process.env.ATLAS_CONNECTIVITY != null,
+    adl: 'unknown', // TODO
```

Comment:
Should we create follow up tickets for these three todos or are these something you will tackle in this PR?

Reply:
Follow ups if we want to keep these. Currently those special environments are kept separate and run on their own variant, there isn't much use to having them stated here, unless we feel otherwise. For example it obvious when we're running against ADL, so is there a need for this?